File: app/app-farma/src/models/order.ts
import { Address, Client } from './client';
import  { Sku } from './product';
import  { PaymentMethodType } from './enumerables';
import  { Store } from './store';

export class Order {
    SkusRequesteds: Array<Sku>;
    AgressReference: boolean;
    AgressSimilar: boolean;
    AgressGeneric: boolean;
    Responses: Array<OrderResponse>;
    RequestedOn: Date;
    ScheduledTo: Date;
    DeliveredOn: Date;
    SelectedPayment: PaymentMethod;
    SelectedDelivery: Delivery;
    AddressToShip: Address;
    RequestedStores: Array<Store>;
    Client: Client;

    constructor()
    {
        this.SkusRequesteds = new Array<Sku>();
    }
}

export class Delivery {
    Price: number;
    MaxDistance: number;
}

export class OrderResponse {
    SkuReplys: Array<OrderResponseProduct>;
    DrugStore : Store;
    Accepted: boolean;
}

export class OrderResponseProduct {
    OriginalSku: Sku;
    OferredSku : Sku;
    Price: number;
    OutStock: boolean;
}

export class PaymentMethod {
    Type: PaymentMethodType;
}

File: service/IhChegou.Repository.Pharma/Model/Orders/Order.cs


using System;

namespace IhChegou.Repository.Pharma.Model.Orders
{
    public class Order
    {
        public virtual int Id { get; set; }
        public virtual bool AgreesReference { get; set; }
        public virtual bool AgreesSimilar { get; set; }
        public virtual bool AgressGeneric { get; set; }
        public virtual DateTime? RequestedOn { get; set; }
        public virtual DateTime? ScheduledTo { get; set; }
        public virtual DateTime? DeliveredOn { get; set; }

        public virtual int? Client_Id { get; set; }
        public virtual int? SelectedPayment_Id { get; set; }
        public virtual int? SelectedDelivery_Id { get; set; }
        public virtual int? AddressToShip_Id { get; set; }


        //public virtual IList<Store> RequestedStores { get; set; }
        //public virtual IList<OrderSkuRequest> SkusRequesteds { get; set; }
        //public virtual IList<OrderResponse> Responses { get; set; }


    }
}